src/device/cuda_intrinsics/synchronization.jl

```diff
@@ -12,6 +13,33 @@ threads in the block.
 """
 @inline sync_threads() = ccall("llvm.nvvm.barrier0", llvmcall, Cvoid, ())
 
+"""
+    sync_threads_count(pred::Int32)
+
+is identical to `__syncthreads()` with the additional feature that it evaluates predicate
+for all threads of the block and returns the number of threads for which predicate evaluates
```

Comment:
Maybe rename the `pred` argument to `predicate` and use backticks in the docstring: "with the additional feature that it evaluate `predicate` for all threads"

Reply:
I changed the name of argument and code-styled the corresponding keywords in the docstring. Also, the documents now starts with "Identical to ..." to match the style of the file.